### PR TITLE
Concat hooks with &&

### PR DESCRIPTION
**Pull request by ggcasuso**
Concat hooks with && to execute following command only if the previous command was successful.

For example if you have two hooks, first one to check your syntax and second one to check your code quality, if syntax fails right now the second one it's going to be executed, with this change if any of the hooks fails, execution stops.

Steps to reproduce:

Add the following hook configuration
```bash
"pre-commit": [
    "echo execution-error;`exit 1`",
    "echo first hook;"
],
```

If you execute `./cghooks pre-commit` output shows:

```bash
execution-error
first hook
```

And with the fix the output will be:

```
execution-error
```

### CODE REVIEW

File: src/helpers.php [hunk 1]
```diff
@@ -57,3 +57,17 @@ function git_dir()
...
+    /**
+     * Return hook contents
+     *
+     * @param array|string $contents
+     *
+     * @return string
+     */
+    function get_hook_contents($contents)
+    {
+        return is_array($contents) ? implode(" && ", $contents) : $contents;
```
Comment (BrainMaestro) [inline src/helpers.php]: I would still like the commands to be on multiple lines. let's use `" && \" . PHP_EOL` as the separator

File: tests/HookCommandTest.php [hunk 2]
```diff
@@ -20,4 +20,23 @@ public function it_tests_hooks_that_exist()
             $this->assertContains(str_replace('echo ', '', $script), $commandTester->getDisplay());
         }
     }
+
+    /**
+     * @test
+     */
+    public function it_not_continue_if_previous_hook_fail()
```
Comment (BrainMaestro) [inline tests/HookCommandTest.php]: typo: let's use `it_terminates_if_previous_hook_fails`

File: tests/HookCommandTest.php [hunk 3]
```diff
@@ -20,4 +20,23 @@ public function it_tests_hooks_that_exist()
...
+                'echo execution-error;exit 1',
+                'echo before-commit'
+            ],
+        ];
+
+        $command = new HookCommand('pre-commit', $hook['pre-commit']);
+        $commandTester = new CommandTester($command);
+
+        $commandTester->execute([]);
+        $this->assertContains('execution-error', $commandTester->getDisplay());
```
Comment (BrainMaestro) [inline tests/HookCommandTest.php]: this is not enough. can you also add 
```php
$this->assertNotContains('before-commit', $commandTester->getDisplay());
```

Comment (ggcasuso) [reply tests/HookCommandTest.php]: You're right, pretty much better tested

File: src/helpers.php (hunk 1 above)
Comment (ggcasuso) [reply src/helpers.php]: Right, .lock will be more readable in multiline

**Comment by BrainMaestro** [comment]
I will try to batch this with the other open PR #105 before I make a new patch release. In the meantime, if you absolutely need this, you can use `dev-master`. Thanks!

**Comment by ggcasuso** [comment]
> I will try to batch this with the other open PR #105 before I make a new patch release. In the meantime, if you absolutely need this, you can use `dev-master`. Thanks!

Thank you!

**Pull request merged**